pytorch_lightning/accelerators/accelerator.py

```diff
@@ -201,7 +199,7 @@ def training_step(
         """
         step_kwargs = self.to_device(step_kwargs)
 
-        with self.precision_plugin.train_step_context(), self.training_type_plugin.train_step_context():
+        with self.training_type_plugin.precision_plugin.train_step_context(), self.training_type_plugin.train_step_context():
```

Comment:
We basically have two different contexts here: One for forward + backward (current train_step_context) and one for forward only.

I think we could change it down to these two as well in this PR. From my perspective there is no difference between forward in training and forward in val/test/predict so we should be able to combine them.